lib/rules/scrollable-region-focusable-matches.js

```diff
@@ -0,0 +1,15 @@
+/**
+ * Note:
+ * `excludeHidden=true` for this rule, thus considering only elements in the accessibility tree.
+ */
+const { isHidden, querySelectorAll } = axe.utils;
+const nodeAndDescendents = querySelectorAll(virtualNode, '*');
+
+const hasVisibleChildren = nodeAndDescendents.filter(
+	({ actualNode }) => !isHidden(actualNode)
+);
+if (!hasVisibleChildren) {
+	return false;
+}
+
+return !!axe.utils.getScroll(node, 13); //1em ~= 13px
```

Comment:
getScroll is much faster than checking if the thing has visible content. You should run this first.